Source/cameramanager.cpp

```diff
@@ -0,0 +1,111 @@
+#include "cameramanager.hpp"
+#include "ui_cameramanager.h"
+#include "editortab.hpp"
+#include <QDebug>
+#include <QFileDialog>
+#include <QMessageBox>
+
+class CameraListItem final : public QListWidgetItem
+{
+public:
+    CameraListItem(QListWidget* pParent, Camera* pCamera)
+        : QListWidgetItem(QString(pCamera->mName.c_str()), pParent), mCamera(pCamera)
+    {
+
+    }
+
+    const Camera* GetCamera() const
+    {
+        return mCamera;
+    }
+private:
+    Camera* mCamera = nullptr;
+};
+
+CameraManager::CameraManager(QWidget *parent, EditorTab* pParentTab, const QPoint* openedPos) :
+    QDialog(parent),
+    ui(new Ui::CameraManager),
```

Comment:
Wouldn't it be more convenient to smart_ptr it?

Reply:
its generated code and blows up QtCreator if you change it